packages/web/src/uiUtils.ts

```diff
@@ -224,3 +224,16 @@ export function scrollIntoViewIfNeeded(element: Element | undefined) {
 
 const kControlCodesRe = '\\u0000-\\u0020\\u007f-\\u009f';
 export const kWebLinkRe = new RegExp('(?:[a-zA-Z][a-zA-Z0-9+.-]{2,}:\\/\\/|www\\.)[^\\s' + kControlCodesRe + '"]{2,}[^\\s' + kControlCodesRe + '"\')}\\],:;.!?]', 'ug');
+
+// flash is retriggered whenever the value changes
+export function useFlash(flash: any | undefined) {
```

Comment:
What's the semantic? Whenever `flash` is truthy it is going to "flash" and unflash in a second? Should this be `flash: boolean` instead?

Reply:
whenever `flash` changes its value, `useFlash` returns true for two seconds. if it were `flash: boolean` then you couldn't retrigger it. it depends on something where it can compare references

Reply:
This API signature should be `const useFlash = (): [boolean, () => void]`

You return a function that, when called, triggers the flash by updating state. You have a `useState` with an incrementing counter to trigger rerenders (i.e. the `useEffect` dependencies array contains the state number).

Reply:
In particular it looks like you want your `setFlash` to take an ID object (we can just use the attachment itself), which is then cleared when the timer finishes. You then compare against that ID object to determine if you want to flash that particular attachment when it changes.

I believe you also trigger your `scrollTo` when you have a matching flash object.

Reply:
How's that different from the current implementation, apart from the placement of the useEffect?

Reply:
```suggestion
export function useFlash(idObject?: any) {
```